File: src/Belajar_SOLID_13_Interface_Segregation_Principle/Java/Main.java
package Belajar_SOLID_13_Interface_Segregation_Principle.Java;

import com.sun.management.VMOption;

interface IPayment{
    void setPaymentName();
    void setAmount();
    void bankId();
    void virtualBankID();
    void accountID();
}

interface EWalletProvider{
    void accountID();
    void walletProviderId();
}

interface PaymentProvider{
    void paymentName();
    void amount();
}

interface BankProvider{
    void bankId();
    void virtualAccount();
}

class GoPay implements EWalletProvider,PaymentProvider{

    @Override
    public void accountID() {

    }

    @Override
    public void walletProviderId() {

    }

    @Override
    public void paymentName() {

    }

    @Override
    public void amount() {

    }
}

class Maindiri implements PaymentProvider,BankProvider{
    @Override
    public void paymentName() {

    }

    @Override
    public void amount() {

    }

    @Override
    public void bankId() {

    }

    @Override
    public void virtualAccount() {

    }
}
//    Dengan memecah interface menjadi beberapa interface kecil,
//    kita dapat dengan mudah menyesuaikan kebutuhan masing-masing class.
//    Kita juga dapat dengan mudah menambahkan class-class baru yang mengimplementasi
//    interface yang sesuai.

File: src/Belajar_SOLID_13_Interface_Segregation_Principle/Kotlin/Main.kt
package Belajar_SOLID_13_Interface_Segregation_Principle.Kotlin

interface IPayment{
    fun setPaymentName();
    fun setAmount();
    fun bankID()
    fun virtualBankID();
    fun accountID()
}

interface EWalletProvider{
    fun accountId();
    fun walletProviderId();
}

interface PaymentProvider{
    fun paymentName();
    fun amount();
}

interface BankProvider{
    fun bankID();
    fun virtualAccount();
}

class Gopay: EWalletProvider,PaymentProvider{
    override fun accountId() {
        TODO("Not yet implemented")
    }

    override fun walletProviderId() {
        TODO("Not yet implemented")
    }

    override fun paymentName() {
        TODO("Not yet implemented")
    }

    override fun amount() {
        TODO("Not yet implemented")
    }
}

class Maindiri: BankProvider,PaymentProvider{
    override fun paymentName() {
        TODO("Not yet implemented")
    }

    override fun amount() {
        TODO("Not yet implemented")
    }

    override fun bankID() {
        TODO("Not yet implemented")
    }

    override fun virtualAccount() {
        TODO("Not yet implemented")
    }
}

class BNI : BankProvider,PaymentProvider{
    override fun paymentName() {
        TODO("Not yet implemented")
    }

    override fun amount() {
        TODO("Not yet implemented")
    }

    override fun bankID() {
        TODO("Not yet implemented")
    }

    override fun virtualAccount() {
        TODO("Not yet implemented")
    }
}